### PR TITLE
Adding RBAC namespace selection for metrics server

### LINKED ISSUE #645: Allow metrics-server non-kube-system RBAC (+ClusterRole) deployment
Hi!

We have a pretty rare case in our company which is still spread across clusters out there. In a situation where you can not deploy to kube-system namespace it is not possible to deploy the chart, because metrics-server needs access to 
extension-apiserver-authentication ConfigMap, thus a RoleBinding should be deployed to kube-system in order to achieve this: https://github.com/kedacore/charts/blob/43b0f516e0d101c18830d06126700916c029712b/keda/templates/metrics-server/clusterrolebinding.yaml#L24 
A possible solution here would be to add a ClusterRole with same permissions as the extension-apiserver-authentication-reader Role and make it optional if kube-system deployment is not possible. I have already forked the chart and am ready to create a MR if it could be approved. Maybe you have other suggestions, would be really glad to hear them.

Looking forward to hearing from you!

Nils.

### PR DESCRIPTION
**Pull request by Rotzke**
_Provide a description of what has been changed_

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)* [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation)
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes #645

### CODE REVIEW

**Comment by Rotzke** [comment]
Hi @JorTurFer ! I have made a duplicate of https://github.com/kedacore/charts/pull/657 because it is stuck since the 28. of June and I was actually the one who requested the feature, please review my MR, I excluded version changes as you requested in the initial MR.

**Comment by JorTurFer** [comment]
To make users 100% aware of the goal, we could call the parameter like `controlPlaneServiceAccountsNamespace`. I know that there are docs where it's explained how to enable the control plane permission segregation that spread the permissions in multiple service account, but I can't find it (it's a control plane argument). If you find it, I think that it's a good idea adding the link

**Comment by Rotzke** [comment]
> To make users 100% aware of the goal, we could call the parameter like `controlPlaneServiceAccountsNamespace`. I know that there are docs where it's explained how to enable the control plane permission segregation that spread the permissions in multiple service account, but I can't find it (it's a control plane argument). If you find it, I think that it's a good idea adding the link

@JorTurFer I have renamed the variables, haven't really found those docs you mentioned, though it is not a command line argument for any of control plane components, I guess. Can we still get this merged into the next release/minor/patch version? Would be really cool, we are shivering in anticipation, waited for some time. Thanks in advance!

**Comment by JorTurFer** [comment]
> Can we still get this merged into the next release/minor/patch version?

Yeah! just let me look for the link a few days more and definitively this will be included for next version

File: keda/README.md [hunk 1]
```diff
@@ -178,6 +178,7 @@ their default values.
 | `podDisruptionBudget.metricServer` | object | `{}` | Capability to configure [Pod Disruption Budget] |
 | `podLabels.metricsAdapter` | object | `{}` | Pod labels for KEDA Metrics Adapter |
 | `podSecurityContext.metricServer` | object | [See below](#KEDA-is-secure-by-default) | [Pod security context] of the KEDA metrics apiserver pod |
+| `rbac.controlPlaneServiceAccountsNamespace` | string | `"kube-system"` | Customize the namespace for metrics-server |
```
Comment (zroubalik) [inline keda/README.md:181]: Could you please add more context here, that we are talking about k8s metrics server.

Comment (Rotzke) [reply keda/README.md]: Have updated, is this OK?

Comment (JorTurFer) [reply keda/README.md]: I've found the docs page :) 
This value is for the namespace where the controller service accounts are place when the control manager flag `--use-service-account-credentials` is set -> https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/.

[Datadog explains it better](https://docs.datadoghq.com/security/default_rules/cis-kubernetes-1.5.1-1.3.3/), but I'd not use their docs but the official k8s repo

Comment (Rotzke) [reply keda/README.md]: Updated again, please take a look.

**Review by zroubalik — APPROVED** [review]

**Review by wozniakjan — APPROVED** [review]

**Pull request merged**